cmd/kube-scheduler/app/options/configfile_others_test.go

```diff
@@ -0,0 +1,24 @@
+//go:build !windows
+// +build !windows
+
+/*
+Copyright 2023 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package options
+
+const (
+	fileNotFound = "no such file or directory"
```

Comment:
Don't do this.

You can compare errors using `errors.Is`. The `os` package provides all errors as constants, such as `os.ErrNotExist`.

FWIIW https://github.com/kubernetes/community/blob/master/sig-scheduling/CONTRIBUTING.md#technical-and-style-guidelines

Reply:
That is not so easy to achieve.

First of all, I tried using:

```
expectedErr:    os.ErrNotExist,
```

instead of using:

```
expectedErr:    fmt.Errorf(fileNotFound),
```

in ``configfile_test.go``, which needs the changes proposed in this PR. However, that will fail:

```
=== RUN   TestLoadConfigFromFile/case_0:_Empty_scheduler_config_file_path
    configfile_test.go:135:
                Error Trace:    C:/Users/Administrator/workdir/kubernetes/cmd/kube-scheduler/app/options/configfile_test.go:135
                Error:          Error "open : The system cannot find the file specified." does not contain "file does not exist"
                Test:           TestLoadConfigFromFile/case_0:_Empty_scheduler_config_file_path
```

That is because ``os.ErrNotExist`` / ``fs.ErrNotExist`` is actually defined in golang's internal source code as such [1]:

```
ErrNotExist   = errors.New("file does not exist")
```

Which is not the error message you would see on Windows.

The ``configfile_test.go`` file which contains those tests is using ``assert.ErrorContains(t, err, test.expectedErr.Error())``, and we could replace it with ``errors.Is`` / ``assert.ErrorIs``, and that would make the test pass, however, it'll make the other 2 tests defined to fail, because ``errors.Is`` / ``assert.ErrorIs`` expects the given errors to be **exactly** the same (===), or if it implements a method Is(error) bool such that Is(target) returns true (which is not always the case) [2]
. This is problematic, since some errors (e.g.: the decode error checked in this test suite [3]) does not implement the ``Is`` method, and it would have to be added. Without this, we can't use ``errors.Is`` / ``assert.ErrorIs``, not even for errors with identical error messages [4].

Thoughts?

[1] [ErrNotExist](https://cs.opensource.google/go/go/+/master:src/internal/oserror/errors.go;bpv=1;bpt=1;l=16?gsn=ErrNotExist&gs=KYTHE%3A%2F%2Fkythe%3A%2F%2Fgo.googlesource.com%2Fgo%3Flang%3Dgo%3Fpath%3Dinternal%2Foserror%23var%2520ErrNotExist)   = [errors](http://godoc.org/errors).[New](https://cs.opensource.google/go/go/+/master:src/errors/errors.go;drc=4956c3437bd2f4448bcec51321f123d03731ddfc;l=61)("file does not exist")
[2] https://pkg.go.dev/errors#Is
[3] https://github.com/kubernetes/kubernetes/blob/022d50fe3a1bdf2386395da7c266fede0c110040/staging/src/k8s.io/apimachinery/pkg/runtime/error.go#L103C6-L103C24
[4] https://pkg.go.dev/errors#New

Reply:
> This is problematic, since some errors (e.g.: the decode error checked in this test suite [3]) does not implement the Is method,

You can easily implement those for the `apimachinery/pkg/runtime` package as part of this effort.